apps/prairielearn/assets/scripts/lib/saveButtonEnabling.ts

```diff
@@ -1,25 +1,21 @@
-import { onDocumentReady } from '@prairielearn/browser-utils';
-import { on } from 'delegated-events';
-
-onDocumentReady(() => {
-  // Select the save button and disable it upon page load.
-  const saveButton = $('#save-button')[0];
+export function saveButtonEnabling(form: HTMLFormElement, saveButton: HTMLButtonElement) {
   saveButton.setAttribute('disabled', 'true');
 
   // Create a record to track if each value has changed. We are not storing initial values, but rather a boolean for each input to track if the value has changed. This is important so we don't have to read all values for each input on every change event when we check if there are differences.
   const valueHasChanged: Record<string, boolean> = {};
-  $('input, select').each(function () {
-    valueHasChanged[this.id] = false;
+  form.querySelectorAll('input, select').forEach((element) => {
+    valueHasChanged[element.id] = false;
   });
```

Comment:
Please check this for a simpler method: https://github.com/PrairieLearn/PrairieLearn/blob/6a1af4a93c80eebcc4348b49ba8389adac152abf/apps/prairielearn/assets/scripts/lib/confirmOnUnload.ts#L1-L10